src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -329,6 +363,48 @@ public Record nextRecord() {
     }
 
     /**
+     * Reports kafka consumer statistics at an interval.
+     */
+    private void reportStatus() {
+        // If we've reported status more recently than 30 seconds we skip over this.
+        if (lastReportedStatusMillis != null && TimeUnit.MILLISECONDS.toSeconds(clock.millis() - lastReportedStatusMillis) < 30) {
+            return;
+        }
+
+        final Map<TopicPartition, Long> topicPartitions = getKafkaConsumer().endOffsets(
+            getKafkaConsumer().assignment()
+        );
+
+        // Capture the ending offset for this topic/partition on the kafka topic
+        for (final Map.Entry<TopicPartition, Long> topicPartition : topicPartitions.entrySet()) {
+            metricsRecorder.assignValue(
+                getClass(),
+                "endOffset.topic." + topicPartition.getKey().topic() + ".partition." + topicPartition.getKey().partition(),
```

Comment:
@Crim should we do topic / partition / metric or metric / topic / partition? I can't decide...

Reply:
My 2 cents.  I'd do topic.partition.metric.  We may have more than one metric listed under there in the future.

Reply:
Like lag below.  We can keep the keyspace lower by putting the related metrics together.